gammapy/stats/fit_statistics.py

```diff
@@ -149,72 +152,62 @@ def wstat(n_on, n_off, alpha, mu_sig, extra_terms=True):
     alpha = np.atleast_1d(np.asanyarray(alpha, dtype=np.float64))
     mu_sig = np.atleast_1d(np.asanyarray(mu_sig, dtype=np.float64))
 
-    mu_bkg = _get_wstat_background(n_on, n_off, alpha, mu_sig)
+    if mu_bkg is None:
+        mu_bkg =  get_wstat_mu_bkg(n_on, n_off, alpha, mu_sig)
 
     term1 = mu_sig + (1 + alpha) * mu_bkg
-    term2 = - n_on * np.log(mu_sig + alpha * mu_bkg)
-    term3 = - n_off * np.log(mu_bkg)
+    term2_ = - n_on * np.log(mu_sig + alpha * mu_bkg)
+    
+    zero_val = np.zeros(len(n_on))
```

Comment:
Is this needed. Can't you just use `0` below and Numpy broadcasting will take care of this because always `n_on` and `n_off` have the same length?